logstash-core/lib/logstash/util/safe_uri.rb

```diff
@@ -14,7 +14,7 @@ class LogStash::Util::SafeURI
   attr_reader :uri
 
   public
-  def initialize(arg)
+  def initialize(arg, requires_host=true)
```

Comment:
Honestly I don't really see a reason to not require a host, thoughts?

Reply:
java.net.URI is pretty open for what it can accept - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URI.html, but I suspect you are almost certainly right, that we don't need it